### PR TITLE
[IMPROVEMENT] Interactive keyboard dismissal improved

### PR DESCRIPTION
**Pull request by Sameesunkaria**
@RocketChat/ios

- The text input bar no longer lags 1 frame behind the keyboard.
- Should have better performance characteristics, since the new implementation is only called when the frame of the keyboard actually changes (unlike previously, where the method was called inside `scrollViewDidScroll`)

### CODE REVIEW

**Comment by rafaelks** [comment]
@Sameesunkaria Thanks for the PR! 👍 

There's one issue with your changes, is that: if I'm scrolling thought the list of messages and then I press the textView to start typing a new message, the scroll goes to the bottom again. It shouldn't change the position if I'm looking into older messages. I understand why it's happening that, but can you do that only if user is at the bottom of the list of messages?

**Comment by Sameesunkaria** [comment]
Oh, I believe the "scroll to bottom when keyboard is shown" behavior was already implemented previously. But I can see how it would get irritating cause now the keyboard slides out automatically. I'll get that fixed!

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -459,10 +463,6 @@ final class ChatViewController: SLKTextViewController {
         }
     }
 
-    override func textViewDidBeginEditing(_ textView: UITextView) {
-        scrollToBottom(true)
-    }
-
```
Comment (Sameesunkaria) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Was rather straight forward.

**Comment by Sameesunkaria** [comment]
@rafaelks It should all work as expected now!

**Comment by Sameesunkaria** [comment]
@rafaelks it would be nice if you could merge this with 2.1.0 beta.

**Comment by rafaelks** [comment]
Looking good! Thank you!!!

**Pull request merged**